util_function_groups/value_replacement.R

```diff
@@ -1,6 +1,35 @@
 
 # replace appearances of originals with replacements inside the vector
 util.recode <- function(vector, originals, replacements){
+  # check if any of the replacement values also appear in the originals.
+  if(any(replacements %in% originals)){
```

Comment:
@sarahgripshover, great catch. This is indeed a very surprising behavior.

However, I'm confused. Do you think this is the desired behavior? I.e., do you think that the vector in your example _should_ turn into all "c"s? It seems not only surprising, but also just downright wrong. If you said you wanted "a" turned into "b", then surely the function is broken if it returns "c" for "a". Right?

Given that this function is -- in my estimation -- broken. The right remedy is not to throw a warning. It's to fix the function. I think the best ways to deal with that would be to:

* Fix it yourself.
* Comment on my initial commit that it's messed up so I can fix it (or whoever was responsible for submitting broken code).
* Use github to start an issue (this is probably the most appropriate).


Anyway, here's my recommended new version of this function, along with a test that would fail using the original version of the function. 

```r
util.recode <- function(vector, originals, replacements){
    # replace appearances of "originals" with "replacements"
    if(length(originals) != length(replacements)){
        "Original and replacement should have equal lenght" %>%
            util.warn()
    }
    if(any(duplicated(originals))){
        "Originals should not include duplicate values" %>%
            util.warn()
    }
    
    new_vec <- vector
    for(v in originals){
        new_vec[vector == v] <- replacements[originals == v]
    }
    new_vec
}

gymnast_test__util.recode <- function(){
    # basic test
    test_name <- "basic replacement"
    vector <- c("a","b","fefe","c")
    x <- c("a","b","c")
    y <- c("A","B","C")
    expectation <- c("A","B","fefe","C")
    if( !identical(expectation,util.recode(vector,x,y)) ){
        util.warn("recode fails: " %+% test_name)
    }
    
    test_name <- "multiple replacement includes originals"
    vector <- c("a","b","fefe","c")
    x <- c("a","b","c")
    y <- c("b","c","d")
    expectation <- c("b","c","fefe","d")
    if( !identical(expectation,util.recode(vector,x,y))){
        util.warn("recode fails: " %+% test_name)
    }
}
```

Finally, I just want to note that you're making changes in a branch that's very much out of date (by 8 days). In fact, the file you're working on no longer exists on master. I made a separate pull request with the code above: 

https://github.com/PERTS/gymnast/pull/7/files